### ISSUE TITLE
cisv(t, A, b) to compute exp(im*A*t)*b for Hermitian A?

### ISSUE THREAD

**Issue by stevengj** [SciML/ExponentialUtilities.jl#176]
It is pretty common to want to compute unitary matrix exponentials of the form $e^{iAt}b$ for Hermitian matrices $A$.   It would be nice to have a method that takes advantage of $A$ being Hermitian here, and right now this doesn't seem to exist:

* $iA$ is anti-Hermitian (skew-Hermitian), so the `ishermitian` flag for `expv` cannot be used.
* the `t` parameter of `expv!` is [constrained to be `Real`](https://github.com/SciML/ExponentialUtilities.jl/blob/4e5be62da033d25a22b9b2a66539861036699f30/src/krylov_phiv.jl#L76) for some reason, so you can't use `expv(im*t, A, b)`

The simplest approach to improving this would just be to allow `t` to be complex (or even an arbitrary `Number`) type.  Offhand, I don't see anything in the algorithm that requires it to be `Real`?   That way, you could immediately take advantage of `A` being Hermitian in [using a tridiagonal eigensolver](https://github.com/SciML/ExponentialUtilities.jl/blob/4e5be62da033d25a22b9b2a66539861036699f30/src/krylov_phiv.jl#L90-L91) as well as in the Krylov procedure ([via Lanczos](https://github.com/SciML/ExponentialUtilities.jl/blob/master/src/arnoldi.jl#L246-L247)).

For dense matrix exponential, Julia exports an [optimized Hermitian method for `cis(A) = exp(im*A)`](https://github.com/JuliaLang/julia/pull/40194).  It would be natural to expose a `cisv(t, A, b)` that computes `exp(im*A*t)*b` as well, and which under the hood calls `exp(im*t, A, b)` to exploit Hermitian `A`.

**Comment by ChrisRackauckas** [comment]
I'm definitely not opposed to it. Someone just needs to implement it. It's only missing because we haven't ran into it yet, but it's a good self-contained project.

**Comment by goerz** [comment]
Allowing `t` to be a complex `Number` would be a prerequisite for using `ExponentialUtilities` as a backend for `QuantumPropagators`:

https://github.com/JuliaQuantumControl/QuantumPropagators.jl/issues/86

Absorbing `1im` in `A` (the Hamiltonian) isn't really an option in that context

**Comment by stevengj** [comment]
Seems like it would be just a few-line change (+ tests) to change `t::Real` to `t::Number`.

**Comment by goerz** [comment]
Yeah, I'll probably try `KrylovKit` first, and then when I get around to `ExponentialUtilities`, I'll make a PR here (unless someone beats me to it)
